code/game/machinery/dance_machine.dm

```diff
@@ -504,3 +515,12 @@
 
 /obj/machinery/disco/immobile/wrench_act()
 	return FALSE
+
+/obj/machinery/disco/chaos_staff
+	restdancing = FALSE
+	anchored = TRUE
+
+/obj/machinery/disco/chaos_staff/Initialize(mapload)
+	. = ..()
+	selection = pick(songs)
+	INVOKE_ASYNC(src, TYPE_PROC_REF(/obj/machinery/disco/chaos_staff, breakitdown))
```

Comment:
```suggestion
	INVOKE_ASYNC(src, PROC_REF(breakitdown))
```
This can probably just be a regular proc ref, no need for a type ref (test it before adding)